### DISCUSSION TITLE
How do I use SSL_OP_IGNORE_UNEXPECTED_EOF in linux

### DISCUSSION THREAD

**Discussion by tomar-gaurav** [openssl/openssl#24810] [Q&A] [answered]
Hello,

I want to know how can I set SSL_OP_IGNORE_UNEXPECTED_EOF on linux(ubuntu) which should be set globally for all the connections.

**Comment by mattcaswell** [comment, marked as answer]
You can do this via the OpenSSL config file. The default config file is called `openssl.cnf` and is stored in OPENSSLDIR (which is a location determined at OpenSSL compile time). `openssl version -d` will give you the location of OPENSSLDIR for your system.

You can set this option via the string "IgnoreUnexpectedEOF". A minimal OpenSSL config file that does this might look like this:


````
config_diagnostics = 1

openssl_conf = default_conf

[ default_conf ]

ssl_conf = ssl_sect

[ssl_sect]

system_default = ssl_default_sect

[ssl_default_sect]
Options = IgnoreUnexpectedEOF
````

**Comment by tomar-gaurav** [comment]
Hey. It didn't worked, still getting unexpected EOF for 2-3% of connections

**Reply by mattcaswell** [reply]
There are several reasons why this might be the case. Ones that immediately spring to mind are:

- Error in your config file
- The application is configured to use some other config file location than the default or the application is not using the system default version of OpenSSL
- The application is explicitly changing or overriding the default options
- The application is using a non-default library context (and so doesn't load the default config file)
